buffer/src/main/java/io/netty/buffer/AbstractByteBuf.java

```diff
@@ -1349,7 +1349,7 @@ public boolean equals(Object o) {
 
     @Override
     public int compareTo(ByteBuf that) {
-        return ByteBufUtil.compare(this, that);
+        return this == that? 0 : ByteBufUtil.compare(this, that);
```

Comment:
nit: whitespace before `?`
```suggestion
        return this == that ? 0 : ByteBufUtil.compare(this, that);
```